### PR TITLE
[TC-3] initial steps and documentation

### PR DESCRIPTION
**Pull request by fw-noel**
[TC-3](https://bcdevex.atlassian.net/browse/TC-3)

Objective:
Getting the initial steps documented out for deploying the application
This should now work to build and deploy on dev.

Unfortunately, we have a CSS issue still to resolve!

[TC-3]: https://bcdevex.atlassian.net/browse/TC-3?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .github/workflows/ci-promote-test.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
+name: OpenShift Build and Deploy to Dev
```
Comment (chelsea-EYDS) [inline .github/workflows/ci-promote-test.yaml:1]: name "Promote to Test" ?

File: .github/workflows/ci-promote-test.yaml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        env:
+          OS_NAMESPACE_SUFFIX: test
+        if: steps.changes.outputs.src == 'true'
+        run: |
+          oc login --token="$AUTH_TOKEN" --server="$CLUSTER"
+          cd "$GITHUB_WORKSPACE"
+          make deployment-update
+  build:
+    name: OpenShift Build & Tag for Deployment
+    environment: dev
```
Comment (chelsea-EYDS) [inline .github/workflows/ci-promote-test.yaml:40]: environment: test

File: .github/workflows/ci-promote-test.yaml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+on:
+  push:
+    tags:
+      - test
+
+env:
+  CLUSTER: https://api.gold.devops.gov.bc.ca:6443
+  AUTH_TOKEN: ${{ secrets.OC_AUTH_TOKEN }}
+  TOOLS_TOKEN: ${{ secrets.OC_TOOLS_TOKEN }}
```
Comment (chelsea-EYDS) [inline .github/workflows/ci-promote-test.yaml:11]: Do we use tools token for each deployment/promotion?

File: DEVOPS.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+This app is currently on [Openshift GOLD](https://console.apps.gold.devops.gov.bc.ca/).
+
+The flow is spread across four namespaces, one for each environment (Dev, Test, Prod), and Tools for builds. As well, the Postgres DB sites in an instance of [Patroni](https://github.com/bcgov/patroni-postgres-container) with [backups](https://github.com/BCDevOps/backup-container) (TODO).
+
+### Deployment
+For deployment in a fresh namespace, there are a number of make commands that should assist in the initial setup of the environment.
+
+First create the Patroni instance. This can be done by running `db-prep` to create the necessary secrets, then `db-create` to create the StatefulSet that pulls from the BCGov image of Patroni.
+
+Note that the latest tag provided from BCGov Patroni uses Postgres 12.4.
```
Comment (chelsea-EYDS) [inline DEVOPS.md:12]: It probably doesn't matter, but I wonder if we should downgrade to postgres 12.4 for our local db ?

File: .github/workflows/ci-promote-test.yaml (hunk 3 above)
Comment (fw-noel) [reply .github/workflows/ci-promote-test.yaml:11]: I need to consider this, but let's keep this for now

File: DEVOPS.md (hunk 4 above)
Comment (fw-noel) [reply DEVOPS.md:12]: I think we should as well!

**Review by chelsea-EYDS — APPROVED** [review]
woohoo!

**Pull request merged**